boa_engine/src/object/jsset.rs

```diff
@@ -76,7 +77,8 @@ impl JsSet {
     {
         match Set::delete(&self.inner.clone().into(), &[value.into()], context)? {
             JsValue::Boolean(bool) => Ok(bool),
-            _ => Err(JsValue::Undefined),
+            // TODO: find a better solution to this, without impacting perf
+            _ => unreachable!("`delete` must always return a bool"),
```

Comment:
I'm not 100% sure about these. Why is this like this?

Reply:
Yeah, the wording is a bit confusing. Reworded it to something more descriptive.